src/vs/base/parts/tree/browser/tree.ts

```diff
@@ -96,10 +96,10 @@ export interface ITree extends Events.IEventEmitter {
 
 	/**
 	 * Collapses several elements.
-	 * Provide no arguments and it will recursively collapse all elements in the tree
+	 * Collapses all elements at the greatest tree depth that has expanded elements.
 	 * The returned promise returns a boolean for whether the elements were collapsed or not.
 	 */
-	collapseAll(elements?: any[], recursive?: boolean): WinJS.Promise;
+	collapseDeepestExpandedLevel(): WinJS.Promise;
```

Comment:
Let's leave `collapseAll` in place, having `collapseDeepestExpandedLevel` as a new method